src/Text/Pandoc/Writers/LaTeX.hs

```diff
@@ -1423,24 +1423,35 @@ citeCommand c p s k = do
   args <- citeArguments p s k
   return $ literal ("\\" <> c) <> args
 
-citeArguments :: PandocMonad m
-              => [Inline] -> [Inline] -> Text -> LW m (Doc Text)
-citeArguments p s k = do
-  let s' = stripLocatorBraces $ case s of
-        (Str t : r) -> case T.uncons t of
-          Just (x, xs)
-            | T.null xs
-            , isPunctuation x -> dropWhile (== Space) r
-            | isPunctuation x -> Str xs : r
-          _ -> s
-        _   -> s
-  pdoc <- inlineListToLaTeX p
-  sdoc <- inlineListToLaTeX s'
-  let optargs = case (isEmpty pdoc, isEmpty sdoc) of
+citeArgumentsList :: PandocMonad m
+              => [Inline] -> [Inline] -> [Text] -> LW m (Doc Text)
+citeArgumentsList _ _ [] = return empty
+citeArgumentsList p s (cid:k) 
+  | k == [] = do  
```

Comment:
`null k` is better.

Reply:
It wouldn't hurt to have more descriptive argument names, or a comment on top of the function indicating what is what.